src/pykeen/metrics/ranking.py

```diff
@@ -201,18 +201,27 @@ def _safe_divide(x: float, y: float) -> float:
     return x / max(y, EPSILON)
 
 
-class ZMetric(RankBasedMetric):
+class DerivedRankBasedMetric(RankBasedMetric):
+    """A derived rank-based metric."""
+
+    base_cls: ClassVar[RankBasedMetric]
+    base: RankBasedMetric
+
+    def __init__(self, **kwargs):
+        """Initialize the derived metric."""
+        self.base = rank_based_metric_resolver.make(self.__class__.base_cls, kwargs)
+        self.factor = 1 if self.base.increasing else -1
```

Comment:
this is only used by z-adjusted arithmetic mean rank

Reply:
if we re-implement the AMRI with the other solution it will be necessary